include/zephyr/drivers/stepper.h

```diff
@@ -0,0 +1,227 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_STEPPER_H
+#define ZEPHYR_INCLUDE_STEPPER_H
+
+/** \addtogroup stepper
+ *  @{
+ */
+
+/**
+ * @file
+ * @brief Public API for Stepper Motor Controller
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Stepper Motor Position Format required while setting or getting stepper motor positions.
+ */
+enum micro_step_resolution {
+	FULL_STEP,
+	MICRO_STEP_2,
+	MICRO_STEP_4,
+	MICRO_STEP_8,
+	MICRO_STEP_16,
+	MICRO_STEP_32,
+	MICRO_STEP_64,
+	MICRO_STEP_128,
+	MICRO_STEP_256,
+};
+
+enum stepper_attribute {
+	/* read/write actual a.k.a reference position in the stepper driver */
```

Comment:
```suggestion
	/** read/write actual a.k.a reference position in the stepper driver */
```

(so that the documentation picks it up)